### DISCUSSION TITLE
Yolo v5/v7

### DISCUSSION THREAD

**Discussion by pfeatherstone** [frgfm/Holocron#230] [Ideas]
Hi, thanks for the fantastic library! I'm particularly excited about the detection stuff, namely yolo loss function and detection trainer. I have a few questions:
- Do you match the yolov4 paper AP results using this repo ? 
- Do you plan on supporting yolo v5 and v7 ?
- Do you plan on supporting Detr models and bipartite matching losses?
- If I were to add augmentation, for example using albumentations, would the best way to use it with the trainer is via the data loaders ? 

Many thanks

**Comment by frgfm** [comment]
Hello @pfeatherstone :wave: 

Thanks for the message! Let me try to answer all of those:
- I haven't been training on COCO because my DL gear is more limited than the paper's authors :sweat_smile: But I do have to fix a few things about my implementation to make it perfect
- Sure :) But I was waiting on this to calm down (meaning that some versions are actually just reimplementations in another high level framework), so I'll review again all the papers and check which ones have significant improvements then proceed to implement them!
- I usually try to focus on architectures that aren't available in PyTorch yet. DETR was created by FAIR, and it's available at least in detectron, so I'm not sure (priority on YOLO). Regarding the loss, yes why not!
- the trainer was created so that you can add any transformations you see fit. It needs a dataloader as arg, and this dataloader, as long as it has a specific output signature, can be anything :+1: 

Let me know if you have any questions!
Cheers :v:

**Reply by pfeatherstone** [reply]
Nice thank you for your response. I look forward to seeing what you do with this repo.

**Comment by pfeatherstone** [comment]
Have you considered using [timm](https://github.com/rwightman/pytorch-image-models) for all the backbones ? Oh by the way, the author of that repo mentioned that he would be adding detection support later this year.

**Reply by frgfm** [reply]
Oh sure!
However, I do try to focus my integration on the most efficient (perf vs. latency/FLOPS) architectures. And I cannot compete with the zoo of Ross haha

Hopefully, I'll finish the detection part by the end of August :)
